Gems/ROS2/Code/Source/RobotImporter/URDF/UrdfParser.cpp

```diff
@@ -13,84 +13,102 @@
 #include <AzCore/Debug/Trace.h>
 #include <AzCore/std/string/string.h>
 #include <console_bridge/console.h>
-#include <urdf_model/model.h>
 
-namespace ROS2
+namespace ROS2::UrdfParser::Internal
 {
-    namespace UrdfParser::Internal
+    void CheckIfCurrentLocaleHasDotAsADecimalSeparator()
     {
-        void CheckIfCurrentLocaleHasDotAsADecimalSeparator()
+        // Due to the fact that URDF parser takes into account the locale information, incompatibility between URDF file locale and
+        // system locale might lead to incorrect URDF parsing. Mainly it affects floating point numbers, and the decimal separator. When
+        // locales are set to system with comma as decimal separator and URDF file is created with dot as decimal separator, URDF parser
+        // will trim the floating point number after comma. For example, if parsing 0.1, URDF parser will parse it as 0.
+        // This might lead to incorrect URDF loading. If the current locale is not a dot (as per standard ROS locale), we warn the user.
+        std::locale currentLocale("");
+        if (std::use_facet<std::numpunct<char>>(currentLocale).decimal_point() != '.')
         {
-            // Due to the fact that URDF parser takes into account the locale information, incompatibility between URDF file locale and
-            // system locale might lead to incorrect URDF parsing. Mainly it affects floating point numbers, and the decimal separator. When
-            // locales are set to system with comma as decimal separator and URDF file is created with dot as decimal separator, URDF parser
-            // will trim the floating point number after comma. For example, if parsing 0.1, URDF parser will parse it as 0.
-            // This might lead to incorrect URDF loading. If the current locale is not a dot (as per standard ROS locale), we warn the user.
-            std::locale currentLocale("");
-            if (std::use_facet<std::numpunct<char>>(currentLocale).decimal_point() != '.')
-            {
-                AZ_Warning(
-                    "UrdfParser", false, "Locale %s might be incompatible with the URDF file content.\n", currentLocale.name().c_str());
-            }
+            AZ_Warning(
+                "UrdfParser", false, "Locale %s might be incompatible with the URDF file content.\n", currentLocale.name().c_str());
         }
+    }
 
-        class CustomConsoleHandler : public console_bridge::OutputHandler
-        {
-        private:
-            std::stringstream console_ss;
+    class CustomConsoleHandler : public console_bridge::OutputHandler
```

Comment:
Actually, I don't think we need to capture the console for SDFormat, since the API for parsing an SDF returns an `sdf::Error` vector which contains the same errors that get sent to the `sdf::Console`

The `ROS2::UrdfParser::Parse` function actually returns those errors to the caller, which would allow them to log it how they choose: http://osrf-distributions.s3.amazonaws.com/sdformat/api/13.2.0/classsdf_1_1SDF__VERSION__NAMESPACE_1_1Root.html